api/handler/v1/guardian_service.go

```diff
@@ -0,0 +1,445 @@
+package v1
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	pb "github.com/odpf/guardian/api/proto/guardian"
+	"github.com/odpf/guardian/appeal"
+	"github.com/odpf/guardian/domain"
+	"github.com/odpf/guardian/policy"
+	"github.com/odpf/guardian/resource"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+type ProtoAdapter interface {
+	FromProviderProto(*pb.Provider) (*domain.Provider, error)
+	FromProviderConfigProto(*pb.ProviderConfig) (*domain.ProviderConfig, error)
+	ToProviderProto(*domain.Provider) (*pb.Provider, error)
+
+	FromPolicyProto(*pb.Policy) (*domain.Policy, error)
+	ToPolicyProto(*domain.Policy) (*pb.Policy, error)
+
+	FromResourceProto(*pb.Resource) *domain.Resource
+	ToResourceProto(*domain.Resource) (*pb.Resource, error)
+
+	FromAppealProto(*pb.Appeal) (*domain.Appeal, error)
+	ToAppealProto(*domain.Appeal) (*pb.Appeal, error)
+	FromCreateAppealProto(*pb.CreateAppealRequest) ([]*domain.Appeal, error)
+	ToApprovalProto(*domain.Approval) (*pb.Approval, error)
+}
+
+type GuardianServiceServer struct {
```

Comment:
I realise our service protos have `Guardian` name in it so we can leave them as it.

But for the one that is written in this repo, like this `GuardianServiceServer`, does it make sense if we change it to GRPCServer instead?

Reply:
Or maybe if it also handles non-grpc, we can use `ServiceServer` to be more generic?